accepted/00080-api-versioning.md

```diff
@@ -0,0 +1,214 @@
+should be used in the following methods too.
+
+##### 2. Check the method existence via introspection
+Before making a call, check if the desired API method-signature combination
+exists. This check can be done using the [existing methods](#intro-calls) or
+with the new [single method introspection](#introduce-a-single-method-introspection-method).
+
+The advantage of this method is that it can warn the user before making the
+actual call (one possible use case would be UI apps built on top of the API).
+
+##### 3. Check the method existence via introspection and check exact product flavor/version
+This is the safest and the most complicated way. In addition to the checks made
+in the previous section, also check the the flavor and the API version. All
+these checks make sure that the API call is present and has the desired
+semantics defined in the product flavor and version.
+
+#### API > Script Examples
+Examples showing usage of the 3 ways of consuming the API described above should
+be added to this section.
+
+
+### Improving the API introspection
+
+These areas of the introspection must be addressed:
+
+#### Error code for non-existing method/signature
+Currently, when calling a non-existing method/signature combination, the API
+reports a fault with code `-1`. As of time of writing this document, this error
+code is used to signal other faults too (see the the `CustomInfoHandler.java`
+file).
+
+This needs to be solved by:
+- visiting the current occurences of `-1` fault code in the existing handlers
+  containing "business methods" (e.g. `CustomInfoHandler.java`) and make them
+  use meaningful codes (see the `exception_ranges.txt` document in the codebase)
+- reserving a new code for the absence of method/signature combination and
+  making the existing XMLRPC code use it
+
+#### Introduce a single method introspection method
+In addition to
+[existing introspection methods](#intro-calls),
+implement a call in the `api` namespace for checking, whether a method with
+signature exists in a namespace:
+`apiCallExists(namespace, method, parameters_varargs)`.
+
+The same can already be achieved (although in a bit more complicated way) with
+already-existing introspection methods, but this method is more convenient.
+
+#### Introduce the API flavor endpoint
+Introduce a new method under the `api` namespace returning the product flavor
+(Uyuni/SUMA).
+
+
+### Bumping the API version
+The version of the API is bumped when a breaking change is introduced that
+cannot tracked by introspection, which involves the following:
+- changing a behavior of an existing method
+- removing or adding a field in a structure accepted by a method / returned by a
+  method
+- changing the fault thrown by a method (changing a fault code, throwing new
+  faults or removing a fault)
+
+All other changes shall be tracked by the introspection methods.
+
+A CI automation job shall be written to ease tracking of such changes.
+
+
```

Comment:
Would you start the bumping from the current version number (25), and just increase it in whatever branch a certain change is committed to?

Reply:
Yes. A sligthly safer alternative would be to bump the version before introducing the policies (just to be on the safe side, in case there have been some breaking changes introduced already), but I don't think it would make such a big difference, especially in the long term.